Assets/Mirror/Runtime/Transport/KCP/MirrorTransport/KcpTransport.cs

```diff
@@ -47,15 +47,15 @@ void Awake()
 
             // TODO simplify after converting Mirror Transport events to Action
             client = new KcpClient(
-                () => OnClientConnected.Invoke(),
-                (message) => OnClientDataReceived.Invoke(message, Channels.DefaultReliable),
-                () => OnClientDisconnected.Invoke()
+                () => ClientConnectedCallback.Invoke(),
```

Comment:
should we do null checking here?  before if there was nobody subscribed, there would be no problem,  but now if there is no callback,  it would throw NRE.

So the question is:  do we want NRE?  or do we want the callback to be optional?
```suggestion
                () => ClientConnectedCallback?.Invoke(),
```

Reply:
in this case we'd pass the actions directly into kcp instead of using these lambda functions, so it would need to check if they are null or not inside